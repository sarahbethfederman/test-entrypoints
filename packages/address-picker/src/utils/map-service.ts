export default class MapService {
  private placesService: google.maps.places.PlacesService = {} as google.maps.places.PlacesService;
  private autocompleteService: google.maps.places.AutocompleteService = new google.maps.places.AutocompleteService();
  private sessionToken: google.maps.places.AutocompleteSessionToken = new google.maps.places.AutocompleteSessionToken();

  constructor(map?: google.maps.Map) {
    this.placesService = new google.maps.places.PlacesService(map || document.createElement('div'));
  }

  /**
   * Map enum to string
   * @param status google.maps.places.PlacesServiceStatus
   */
  private apiRejectionReason(status: google.maps.places.PlacesServiceStatus): Error {
    if (status === google.maps.places.PlacesServiceStatus.ZERO_RESULTS) {
      return new Error('ZERO_RESULTS');
    } else if (status === google.maps.places.PlacesServiceStatus.INVALID_REQUEST) {
      return new Error('INVALID_REQUEST');
    } else if (status === google.maps.places.PlacesServiceStatus.OVER_QUERY_LIMIT) {
      return new Error('OVER_QUERY_LIMIT');
    } else if (status === google.maps.places.PlacesServiceStatus.NOT_FOUND) {
      return new Error('NOT_FOUND');
    } else if (status === google.maps.places.PlacesServiceStatus.REQUEST_DENIED) {
      return new Error('REQUEST_DENIED');
    } else if (status === google.maps.places.PlacesServiceStatus.UNKNOWN_ERROR) {
      return new Error('UNKNOWN_ERROR');
    }
    return new Error('UNKNOWN_ERROR');
  }

  public placePredictions(
    input: string,
    componentRestrictions: google.maps.places.ComponentRestrictions
  ): Promise<google.maps.places.AutocompletePrediction[]> {
    return new Promise((resolve, reject) => {
      (this.autocompleteService as google.maps.places.AutocompleteService).getPlacePredictions(
        {
          input,
          sessionToken: this.sessionToken,
          componentRestrictions,
        },
        (suggestions: google.maps.places.AutocompletePrediction[], status: google.maps.places.PlacesServiceStatus) => {
          if (status === google.maps.places.PlacesServiceStatus.OK && suggestions !== null) {
            resolve(suggestions);
          }
          reject(this.apiRejectionReason(status));
        }
      );
    });
  }

  public placeDetails(
    placeId: string
  ): Promise<{
    geometry: google.maps.places.PlaceGeometry | undefined;
    addressComponents: google.maps.GeocoderAddressComponent[];
  }> {
    return new Promise((resolve, reject) => {
      (this.placesService as google.maps.places.PlacesService).getDetails(
        {
          placeId,
          sessionToken: this.sessionToken,
        },
        (
          { geometry, address_components = [] }: google.maps.places.PlaceResult,
          status: google.maps.places.PlacesServiceStatus
        ) => {
          if (status === google.maps.places.PlacesServiceStatus.OK && address_components.length > 0) {
            resolve({
              geometry,
              addressComponents: address_components,
            });
          }
          reject(this.apiRejectionReason(status));
        }
      );
    });
  }
}
