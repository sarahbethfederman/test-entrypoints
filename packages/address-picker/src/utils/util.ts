import { AddressObject } from '../components/Modal';
import { STREET_TYPE } from '../components/Modal/constants';
import * as to from 'to-case';

/**
 * Where possible this function takes the users input and
 * Google's address information and enhances it with more
 * accurate subpremise/unit information
 */
export const transformGoogleResponse = (
  label: string,
  res: google.maps.GeocoderAddressComponent[],
  regionSearchOnly?: boolean
): AddressObject => {
  label = label.toLowerCase();

  // pick the easily determined properties from response
  const { short_name: postcode = '' } = res.find((curr) => curr.types.includes('postal_code')) || {};
  let { short_name: state = '' } = res.find((curr) => curr.types.includes('administrative_area_level_1')) || {};
  let { long_name: streetName = '' } = res.find((curr) => curr.types.includes('route')) || {};
  let { short_name: streetNumber = '' } = res.find((curr) => curr.types.includes('street_number')) || {};
  const { long_name: suburb = '' } = res.find((curr) => curr.types.includes('locality')) || {};
  const { long_name: country = '' } = res.find((curr) => curr.types.includes('country')) || {};
  let unit = '';

  // Google has no concept of "street". See if we recognise a streetType from
  // the route then use that to generate streetName and streetType
  // streetType default to 'street' if regionSearchOnly is false
  const street = streetName.trim().split(' ');
  const streetTypeWith = street[street.length - 1];
  let streetType = !regionSearchOnly ? 'Street' : '';
  if (STREET_TYPE.find((street_type) => street_type.label === streetTypeWith)) {
    streetType = streetTypeWith;
  }

  streetName = streetName.replace(streetType, '').trim();

  // edge case, state is listed as JBT for jervis bay when it should be ACT. Naughty Google
  if (state === 'JBT') {
    state = 'ACT';
  }

  // check for unit / subpremise info by crossreferencing against user input
  const streetNumberData = label.substring(0, label.indexOf(streetName.toLowerCase())).trim();
  if (streetNumberData !== streetNumber) {
    if (streetNumberData.includes('/')) {
      unit = streetNumberData.substring(0, streetNumberData.indexOf('/'));
    } else {
      streetNumber = streetNumberData;
    }
  }

  return {
    unit,
    lotSection: '',
    level: '',
    buildingName: '',
    streetNumber,
    streetName,
    streetType,
    suburb,
    postcode,
    state,
    country,
  };
};

export const getFormatedString = (address: AddressObject) => {
  let formatString = address.unit ? ''.concat(address.unit, '/') : '';
  formatString = address.lotSection ? formatString.concat('Lot. ', address.lotSection) : formatString;
  formatString = address.level ? formatString.concat(' level ', address.level) : formatString;
  formatString = address.buildingName ? formatString.concat(' ', to.capital(address.buildingName), ' ') : formatString;
  formatString = address.streetNumber ? formatString.concat(address.streetNumber) : formatString;
  formatString = address.streetName ? formatString.concat(' ', to.capital(address.streetName)) : formatString;
  formatString = address.streetType ? formatString.concat(' ', to.capital(address.streetType)) : formatString;
  formatString = address.suburb ? formatString.concat(', ', to.capital(address.suburb)) : formatString;
  formatString = address.state ? formatString.concat(', ', to.upper(address.state)) : formatString;
  formatString = address.postcode ? formatString.concat(' ', address.postcode) : formatString;
  formatString = address.country ? formatString.concat(', ', to.capital(address.country)) : formatString;
  return formatString;
};
