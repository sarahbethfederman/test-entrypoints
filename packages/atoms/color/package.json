{
  "name": "@auscred/color",
  "version": "0.0.1",
  "source": "src/Color.tsx",
  "main": "dist/Colour.js",
  "types": "dist/Colour.d.js",
  "files": [
    "lib"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
  