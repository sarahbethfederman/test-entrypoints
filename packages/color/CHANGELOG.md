2.0.2 - New Changelog
