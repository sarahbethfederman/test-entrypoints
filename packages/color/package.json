{
  "name": "@lendi-ui/color",
  "version": "3.0.0",
  "description": "Color utilities for color.",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^6.0.0",
    "styled-components": "^3"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^2.0.0"
  },
  "scripts": {
    "build:esm": "tsc --project tsconfig.esm.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:esm"
  }
}
