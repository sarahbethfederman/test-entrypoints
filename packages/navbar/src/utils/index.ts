export const browserSupportsPassiveListeners = () => {
  let supportsPassive = false;
  try {
    var opts = Object.defineProperty({}, 'passive', {
      get: function() {
        supportsPassive = true;
      },
    });
    // for testing browser compatibility, no need to worry
    // about the null TS issue.
    // @ts-ignore
    window.addEventListener('testPassive', null, opts);
    // @ts-ignore
    window.removeEventListener('testPassive', null, opts);
  } catch (e) {}

  return supportsPassive;
};
