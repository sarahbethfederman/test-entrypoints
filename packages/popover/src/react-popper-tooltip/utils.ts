import * as React from 'react';

export const TooltipContext = React.createContext({});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type Fn = (...args: any[]) => void;
type FnOrBool = undefined | Fn;

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const callAll = (...fns: FnOrBool[]) => (...args: any[]) => fns.forEach((fn) => fn && fn(...args));

export const noop = () => {};

export const canUseDOM = () => !!(typeof window !== 'undefined' && window.document && window.document.createElement);
