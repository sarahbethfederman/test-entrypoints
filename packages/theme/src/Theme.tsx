import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
