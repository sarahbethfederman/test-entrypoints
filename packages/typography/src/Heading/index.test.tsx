import * as React from 'react';
import { mount } from 'enzyme';
import Theme, { Colors, theme } from '@lendi-ui/theme';
import { color as getColor } from '@lendi-ui/color';
import { Heading, HeadingSize, HeadingAlignment } from '../index';
import { deriveSize } from '@lendi-ui/utils';

const sizes: HeadingSize[] = ['xl', 'lg', 'md', 'sm', 'xs', 'xxs'];
const colors: Colors[] = ['primary.500', 'shade.0', 'shade.1000'];
const alignments: HeadingAlignment[] = ['left', 'center', 'right', 'justify'];

let wrapper;
const render = (props) => {
  wrapper = mount(
    <Theme>
      <Heading {...props} />
    </Theme>
  );
};

describe('Heading', () => {
  sizes.forEach((size, index) => {
    it(`should render styles for size "${size}"`, () => {
      render({ size });
      expect(wrapper.find(Heading)).toMatchSnapshot();
    });

    it(`should render tag for size "${size}"`, () => {
      render({ size });
      expect(wrapper.find(Heading).find(`h${index + 1}`)).toHaveLength(1);
    });
  });

  colors.forEach((color) => {
    it(`should render styles for color "${color}"`, () => {
      render({ size: 'xl', color });
      expect(wrapper.find(Heading)).toHaveStyleRule('color', color && getColor(color)({ theme }));
    });
  });

  alignments.forEach((alignment) => {
    it(`should render styles for align ${alignment}`, () => {
      render({ size: 'xl', align: `${alignment}` });
      expect(wrapper.find(Heading)).toHaveStyleRule('text-align', `${alignment}`);
    });
  });

  it('should render styles for margin', () => {
    render({ size: 'xl', m: 'md' });
    expect(wrapper.find(Heading)).toHaveStyleRule('margin', `${deriveSize(1.5)}`);
  });

  describe('test native props and Standard HTML Attributes', () => {
    it('should mount with Aria attributes', () => {
      const ARIA_LABEL = 'testLabel';
      const ARIA_DESCRIBE_BY = 'info';
      render({
        'aria-label': ARIA_LABEL,
        'aria-describedby': ARIA_DESCRIBE_BY,
      });
      const attributes = wrapper.find(Heading).props();
      expect(attributes['aria-label']).toBe(ARIA_LABEL);
      expect(attributes['aria-describedby']).toBe(ARIA_DESCRIBE_BY);
    });
    it('should mount with native props like id, itemId', () => {
      const TEXT_ID = 'testId';
      render({
        id: TEXT_ID,
        itemID: 'itemId',
      });
      const attributes = wrapper.find(Heading).props();
      expect(attributes.id).toBe(TEXT_ID);
      expect(attributes['itemID']).toBe('itemId');
    });
  });
});
