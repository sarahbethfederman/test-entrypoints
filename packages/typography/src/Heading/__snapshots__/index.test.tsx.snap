// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render styles for size "lg" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(2.3125 * var(--lendi-ui-size));
  font-weight: 400;
  line-height: calc(40 / 37);
  font-family: 'Cabin',sans-serif;
}

<Styled(HeadingTag)
  size="lg"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
    margin-top: ",
            "0",
            ";
    margin-bottom: ",
            "0",
            ";
  ",
            "
  ",
            [Function],
            "
  ",
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
            ";
",
          ],
        },
        "displayName": "Styled(HeadingTag)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    size="lg"
  >
    <HeadingTag
      className="c0"
      size="lg"
    >
      <h2
        className="c0"
        size="lg"
      />
    </HeadingTag>
  </StyledComponent>
</Styled(HeadingTag)>
`;

exports[`Heading should render styles for size "md" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(1.75 * var(--lendi-ui-size));
  font-weight: 400;
  line-height: calc(32 / 28);
  font-family: 'Cabin',sans-serif;
}

<Styled(HeadingTag)
  size="md"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
    margin-top: ",
            "0",
            ";
    margin-bottom: ",
            "0",
            ";
  ",
            "
  ",
            [Function],
            "
  ",
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
            ";
",
          ],
        },
        "displayName": "Styled(HeadingTag)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    size="md"
  >
    <HeadingTag
      className="c0"
      size="md"
    >
      <h3
        className="c0"
        size="md"
      />
    </HeadingTag>
  </StyledComponent>
</Styled(HeadingTag)>
`;

exports[`Heading should render styles for size "sm" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(1.375 * var(--lendi-ui-size));
  font-weight: 400;
  line-height: calc(28 / 22);
  font-family: 'Cabin',sans-serif;
}

<Styled(HeadingTag)
  size="sm"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
    margin-top: ",
            "0",
            ";
    margin-bottom: ",
            "0",
            ";
  ",
            "
  ",
            [Function],
            "
  ",
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
            ";
",
          ],
        },
        "displayName": "Styled(HeadingTag)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    size="sm"
  >
    <HeadingTag
      className="c0"
      size="sm"
    >
      <h4
        className="c0"
        size="sm"
      />
    </HeadingTag>
  </StyledComponent>
</Styled(HeadingTag)>
`;

exports[`Heading should render styles for size "xl" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(2.875 * var(--lendi-ui-size));
  font-weight: 700;
  line-height: calc(50 / 46);
  font-family: 'Cabin',sans-serif;
}

<Styled(HeadingTag)
  size="xl"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
    margin-top: ",
            "0",
            ";
    margin-bottom: ",
            "0",
            ";
  ",
            "
  ",
            [Function],
            "
  ",
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
            ";
",
          ],
        },
        "displayName": "Styled(HeadingTag)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    size="xl"
  >
    <HeadingTag
      className="c0"
      size="xl"
    >
      <h1
        className="c0"
        size="xl"
      />
    </HeadingTag>
  </StyledComponent>
</Styled(HeadingTag)>
`;

exports[`Heading should render styles for size "xs" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(1.125 * var(--lendi-ui-size));
  font-weight: 400;
  line-height: calc(24 / 18);
  font-family: 'Cabin',sans-serif;
}

<Styled(HeadingTag)
  size="xs"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
    margin-top: ",
            "0",
            ";
    margin-bottom: ",
            "0",
            ";
  ",
            "
  ",
            [Function],
            "
  ",
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
            ";
",
          ],
        },
        "displayName": "Styled(HeadingTag)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    size="xs"
  >
    <HeadingTag
      className="c0"
      size="xs"
    >
      <h5
        className="c0"
        size="xs"
      />
    </HeadingTag>
  </StyledComponent>
</Styled(HeadingTag)>
`;

exports[`Heading should render styles for size "xxs" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: calc(0.875 * var(--lendi-ui-size));
  font-weight: 400;
  line-height: calc(20 / 14);
  font-family: 'Cabin',sans-serif;
}

<Styled(HeadingTag)
  size="xxs"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
    margin-top: ",
            "0",
            ";
    margin-bottom: ",
            "0",
            ";
  ",
            "
  ",
            [Function],
            "
  ",
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
  ",
            [Function],
            "
",
            ";
",
          ],
        },
        "displayName": "Styled(HeadingTag)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    size="xxs"
  >
    <HeadingTag
      className="c0"
      size="xxs"
    >
      <h6
        className="c0"
        size="xxs"
      />
    </HeadingTag>
  </StyledComponent>
</Styled(HeadingTag)>
`;
